app/models/agents/ftpsite_agent.rb

```diff
@@ -32,42 +50,64 @@ class FtpsiteAgent < Agent
     MD
 
     def working?
-      event_created_within?(interpolated['expected_update_period_in_days']) && !recent_error_logs?
+      if interpolated['mode'] == 'read'
+        event_created_within?(interpolated['expected_update_period_in_days']) && !recent_error_logs?
+      else
+        received_event_without_error?
+      end
     end
 
     def default_options
       {
+        'mode' => 'read',
         'expected_update_period_in_days' => "1",
         'url' => "ftp://example.org/pub/releases/",
         'patterns' => [
           'foo-*.tar.gz',
         ],
         'after' => Time.now.iso8601,
+        'force_encoding' => '',
+        'filename' => '',
+        'data_path' => '$.data'
       }
     end
 
     def validate_options
       # Check for required fields
       begin
-        url = options['url']
+        url = interpolated['url']
```

Comment:
We do, but since `url` is not interpolated per event the `{%` tags have to be interpolatable  without additional data. You can use `{% credential name %}`, or `{% if true %}example.com{% endif %}`, but the result has to be a valid URI.